File: frontend/src/objects/rating/request/AppendRatingContestRequest.ts
import type { ContestSlug } from '@/objects/contest/ContestSlug'

/** 追加 rating 比赛请求体；m 为 rating 公式参数，由后端解释。 */
export type AppendRatingContestRequest = {
  contestSlug: ContestSlug
  m: number
}

File: backend/src/main/scala/domains/rating/objects/request/AppendRatingContestRequest.scala
package domains.rating.objects.request

import domains.contest.objects.ContestSlug
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 追加评分比赛请求体，指定比赛 slug 和本场替换粒子数量 m。 */
final case class AppendRatingContestRequest(
  contestSlug: ContestSlug,
  m: Int
)

/** 提供追加评分比赛请求体 JSON codec。 */
object AppendRatingContestRequest:
  given Encoder[AppendRatingContestRequest] = deriveEncoder[AppendRatingContestRequest]
  given Decoder[AppendRatingContestRequest] = deriveDecoder[AppendRatingContestRequest]
